float[] near_pt_1
float[] far_pt_1
float[] near_pt_2
float[] far_pt_2
float[] prob_successful_passage